src/doc/en/developer/coding_in_python.rst

```diff
@@ -585,6 +585,53 @@ by
         return initialize_big_data()
 
 
+Static typing
+=============
+
+Python libraries are increasingly annotated with static typing information;
+see the `Python reference on typing <https://docs.python.org/3/library/typing.html>`.
+
+For typechecking the Sage library, the project uses :ref:`pyright <section-tools-pyright>`;
+it automatically runs in the GitHub Actions CI and can also be run locally.
+
+As of Sage 10.2, the Sage library only contains a minimal set of such type
+annotations. Pull requests that add more annotations are generally welcome.
+
+The Sage library makes very extensive use of Cython (see chapter :ref:`_chapter-cython`).
+Although Cython source code often declares static types for the purpose of
+compilation to efficient machine code, this typing information is unfortunately
+not visible to static checkers such as Pyright. It is necessary to create `type stub
+files (".pyi") <https://github.com/microsoft/pyright/blob/main/docs/type-stubs.md>`_
```

Comment:
I don't know if we want to recommend this. There is https://github.com/cython/cython/pull/5744 which adds auto-gen of type stubs to cython. I would wait for how this PR is developing.
Otherwise there are a few other similar solutions: https://github.com/cython/cython/pull/3818#issuecomment-1666687625

Reply:
Thanks for these pointers!

We could add some of these links to the text.

Reply:
Yes, but then people might actually add auto-genenerted stub files for cython which will become obsolete/incompatible soon with the hopefully soon-added functionality in cython. I would wait 1 or 2 more months and see how the PR develops.

Reply:
OK. I've added a guarded link to the paragraph